azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java

```diff
@@ -7,36 +7,42 @@
 
 import com.azure.core.credential.AccessToken;
 import com.azure.core.credential.TokenCredential;
-import com.azure.core.management.AzureEnvironment;
+import com.azure.core.credential.TokenRequestContext;
 import com.azure.identity.implementation.util.ScopeUtil;
 import com.microsoft.aad.adal4j.AuthenticationContext;
 import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.azure.toolkit.lib.auth.TenantCredential;
 import com.microsoft.azure.toolkit.lib.auth.exception.LoginFailureException;
+import lombok.AllArgsConstructor;
 import org.apache.commons.lang3.StringUtils;
 import reactor.core.publisher.Mono;
 
 import java.net.MalformedURLException;
 import java.time.OffsetDateTime;
 import java.time.ZoneOffset;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
+import java.util.Map;
+import java.util.concurrent.*;
 
-public class RefreshTokenCredentialFactory {
-    public static TokenCredential fromRefreshToken(AzureEnvironment env, String clientId, String tenantId, String refreshToken) {
-        return buildRefreshTokenCredential(env, clientId, tenantId, refreshToken);
+@AllArgsConstructor
+public class RefreshTokenTenantCredential extends TenantCredential {
```

Comment:
migrate TenantCredential to `CredentialManager`
and add `listTenants` method into `CredentialManager`

Reply:
will refactor later